### PR TITLE
raufha_codeReview

### PR DESCRIPTION
**Pull request by SofwareDude**
Analysis of the program.
Was the program available in UC Github on time?
Yes
Is the program documented/commented well enough for you to understand?
Yes
Does the program compile?
Yes
The rationale behind your changes.
Changed the wording according to Kotlin conventions.
Added access modifiers. 
Removed logic that was redundant.

Links to three specific commits that you made to your own group's GitHub repository before the end of the sprint deadline. 
https://github.com/Smoofington/FindMyRecyclingV32001/commit/6d86792198f6d49969243b4260d04741d4400eda
https://github.com/Smoofington/FindMyRecyclingV32001/commit/783253d015ca51e59c498d8ec1d8156e866106d0
https://github.com/Smoofington/FindMyRecyclingV32001/commit/385b8a95bf802ca1a02a40ed6b07b35373e19261

three specific technical concepts that you learned from this code review
Visability modifiers, Variable naming conventions, Class naming convention, debugging with the help of IDE.

### CODE REVIEW

File: app/src/main/java/com/ucmobiledevelopment/freeways/service/IncidentService.kt [hunk 1]
```diff
@@ -22,33 +22,31 @@ class IncidentService : IIncidentService {
...
 
             val service = RetrofitClientInstance.retrofitInstance?.create(IIncidentDAO::class.java)
             val incidentListDTO =
                 async { service?.getAllIncidents(fromCaseYear, toCaseYear, state, county) }
-            var result = incidentListDTO.await()?.awaitResponse()?.body()
+            val result = incidentListDTO.await()?.awaitResponse()?.body()
 
             if (result != null) {
-                if(result.Results[0] != null){
-                    result.Results[0].forEach{
```
Comment (SofwareDude) [inline app/src/main/java/com/ucmobiledevelopment/freeways/service/IncidentService.kt:34]: Removed this part of logic as the result will always be true. Just add result.Result[0].forEach{...} (The IDE highlighted this)

File: app/src/main/java/com/ucmobiledevelopment/freeways/service/IncidentService.kt [hunk 2]
```diff
@@ -22,33 +22,31 @@ class IncidentService : IIncidentService {
     override suspend fun fetchIncidents(fromCaseYear: Int, toCaseYear: Int, state: Int, county: Int) : List<Incident>? {
 
         return withContext(Dispatchers.IO) {
-            var incidentList: MutableList<Incident> = mutableListOf()
```
Comment (SofwareDude) [inline app/src/main/java/com/ucmobiledevelopment/freeways/service/IncidentService.kt:25]: changed var to val as the value never changes. https://kotlinlang.org/docs/coding-conventions.html#property-names

File: app/src/main/java/com/ucmobiledevelopment/freeways/service/IncidentService.kt [hunk 3]
```diff
@@ -22,33 +22,31 @@ class IncidentService : IIncidentService {
...
+            val incidentList: MutableList<Incident> = mutableListOf()
 
             val service = RetrofitClientInstance.retrofitInstance?.create(IIncidentDAO::class.java)
             val incidentListDTO =
                 async { service?.getAllIncidents(fromCaseYear, toCaseYear, state, county) }
-            var result = incidentListDTO.await()?.awaitResponse()?.body()
```
Comment (SofwareDude) [inline app/src/main/java/com/ucmobiledevelopment/freeways/service/IncidentService.kt:30]: Changed var to val as the value never changes.

File: app/src/main/java/com/ucmobiledevelopment/freeways/dto/IncidentDTO.kt [hunk 4]
```diff
@@ -1,19 +1,19 @@
 package com.ucmobiledevelopment.freeways.dto
 
 data class IncidentDTO(
-    val CITY: String? = "",
```
Comment (SofwareDude) [inline app/src/main/java/com/ucmobiledevelopment/freeways/dto/IncidentDTO.kt:4]: Added the internal keyword. https://kotlinlang.org/docs/visibility-modifiers.html#modules

File: app/src/main/java/com/ucmobiledevelopment/freeways/dto/IncidentDTO.kt [hunk 5]
```diff
@@ -2,15 +2,15 @@ package com.ucmobiledevelopment.freeways.dto
...
+    val CASE_YEAR: String? = "",
     val FATALS: String? = "",
     val LATITUDE: String? = "",
     val LONGITUDE: String? = "",
     val STATE: String? = "",
-    val STATENAME: String? = "",
+    val STATE_NAME: String? = "",
     val ST_CASE: String? = "",
     val TOTALVEHICLES: String? = "",
     val TWAY_ID: String? = "",
```
Comment (SofwareDude) [inline app/src/main/java/com/ucmobiledevelopment/freeways/dto/IncidentDTO.kt]: Added a comma for future additions. https://kotlinlang.org/docs/coding-conventions.html#trailing-commas

File: app/src/main/java/com/ucmobiledevelopment/freeways/dto/IncidentDTO.kt (hunk 4 above)
Comment (discospiff) [reply app/src/main/java/com/ucmobiledevelopment/freeways/dto/IncidentDTO.kt:4]: This duplicates a previous review.  When that occurs, first in wins.

File: app/src/main/java/com/ucmobiledevelopment/freeways/service/IncidentService.kt (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/com/ucmobiledevelopment/freeways/service/IncidentService.kt:25]: This duplicates a previous review. When that occurs, first in wins.

File: app/src/main/java/com/ucmobiledevelopment/freeways/service/IncidentService.kt [hunk 6]
```diff
@@ -22,33 +20,31 @@ class IncidentService : IIncidentService {
...
-                        newIncident.longitude = it.LONGITUD ?: ""
-                        newIncident.caseId =  if(it.ST_CASE != null) it.ST_CASE.toInt() else 0
-                        newIncident.way1 = it.TWAY_ID ?: ""
-                        newIncident.way2 = it.TWAY_ID2 ?: ""
-                        newIncident.vehiclesInvolved =  if(it.TOTALVEHICLES != null) it.TOTALVEHICLES.toInt() else 0
-
-                        incidentList.add(newIncident)
-
-                    }
+                result.Results[0].forEach{
```
Comment (discospiff) [inline app/src/main/java/com/ucmobiledevelopment/freeways/service/IncidentService.kt:31]: The forEach is good, though you'll still need a null check on result, as that appears to be a nullable type.

**Comment by discospiff** [comment]
From this review, I'd harvest the forEach.  And, see my note.
The other suggestions duplicate a previous review.  Thus, no need to merge.

**Pull request closed without merging**